src/database/import.ts

```diff
@@ -190,3 +203,35 @@ export default class DatabaseImporter {
     return [root, key].join("/").replace("//", "/");
   }
 }
+
+/**
+ * Batches chunked JSON data up to the specified byte size limit. The emitted batch is an object
+ * containing chunks keyed by pathname.
+ */
+class BatchChunks extends stream.Transform {
+  private batch: Batch = {};
+  private size = 0;
+
+  constructor(private maxSize: number, opts?: stream.TransformOptions) {
+    super({ ...opts, objectMode: true });
+  }
+
+  _transform(chunk: Data, _: BufferEncoding, callback: stream.TransformCallback) {
+    const chunkSize = JSON.stringify(chunk.json).length;
+    if (this.size + chunkSize > this.maxSize) {
+      this.push(this.batch);
+      this.batch = {};
+      this.size = 0;
+    }
+    this.batch[chunk.pathname] = chunk.json;
+    this.size += chunkSize;
```

Comment:
`chunkSize` don't include `chunk.pathname`.
I worry that this can still hit our 10MB incoming request limit. Let's include `pathname.length` as well to be conservative.

10MB may not be the most optimal chunk size. Can we set a lower default number? 1MB is already quite sizable.

Reply:
Done.